### ISSUE TITLE
Safari fetch error not sent (NullError instead)

### ISSUE THREAD

**Issue by semmel** [appsignal/appsignal-javascript#583]
Flightmode (offline) [`fetch()`](https://developer.mozilla.org/en-US/docs/Web/API/fetch) errors on *Safari* in are not sent to the Appsignal servers. Instead the default `NullError` gets posted. 

**To Reproduce**
1. Connect iPhone via cable to a Mac
2. Open any page in Safari (e.g. https://example.com)
3. Open Developer Console on the Mac
4. Switch iPhone in flightmode (make sure WLAN is disconnected too)
6. Run a `fetch` from Dev Console recording the error: `fetch("https://jsonplaceholder.typicode.com/todos/1").catch(e => { globalThis.fetchError = e; })`
7. Quickly switch off flightmode (and reconnect WLAN). Otherwise the Dev Console might close.
8. Inspect if the generated error has `.stack` or `.stacktrace` properties: `fetchError.stack`
9. Because  `fetchError` has no `.stack` or `.stacktrace` properties, `isError(fetchError)` in `/core/src/utils/error.ts` yields `false` and thus it won't be uploaded by `appsignal.sendError(fetchError)`. 

```javascript
fetchError.message // "Load failed"
fetchError.name     // "TypeError"
fetchError.stack     // undefined
fetchError.stacktrace    // undefined
```

**Expected behavior**
Appsignal should treat errors with an `undefined` `.stack` as empty string "" `.stack` and also upload such errors to Appsignal. Just leaving the stacktrace empty.

**Screenshots**
If applicable, add screenshots to help explain your problem. A link to a reproducible test case in [CodeSandbox](https://codesandbox.io) is even better!

**Desktop (please complete the following information):**
 - OS: mac OS 12.6
 - Safari 16.1
 - 
**Smartphone (please complete the following information):**
 - Device: iPhone 13
 - OS: 16
 - Browser safari
 - Version 16.1.1

**Comment by tombruijn** [comment]
Thanks for the report @semmel! We'll have a look at it. We can probably do a null/undefined check somewhere to avoid this error.

**Comment by luismiramirez** [comment]
Hi @semmel! You can find the fix in `@appsignal/core@1.1.19`
